File: online-judge-solutions/Codeforces/1472B.py
#https://codeforces.com/problemset/problem/1472/B
#Lang		:	Python 3.8
#Time		:	108 ms
#Memory     :	0 KB
t=int(input())
for x in range(0,t):
    n=int(input())
    a=list(map(int,input().split()))[:n]
    suma=sum(a)
    n1=a.count(1)
    if(n%2==0):
        print("YES") if suma%2==0 else print("NO")
    else:
        print("YES") if (n1%2==0 and n1>0) else print("NO")

File: online-judge-solutions/Codeforces/1472B.c
//https://codeforces.com/problemset/problem/1472/B
//Lang		:	GNU C11
//Time		:	31 ms
//Memory	:	0 KB
#include <stdio.h>
int main(){
	int t,n,a;
	scanf("%d",&t);
	while(t--){
		scanf("%d",&n);
		int n1=0,sum=0;
		for(int i=0;i<n;i++){
			scanf("%d",&a);
			if(a==1) n1++;
			sum+=a;
		}
		if(n%2==0){
			sum%2==0 ? printf("YES\n"):printf("NO\n");
		}else{
			n1%2==0 && n1>0 ? printf("YES\n"):printf("NO\n");
		}
	}
}
